README.md

```diff
@@ -2,7 +2,74 @@
 
 # Radix tree
 
-Read more about the use cases in the package's docs.
+Radix tree is defined in Wikipedia: <https://en.wikipedia.org/wiki/Radix_tree>
+
+A data structure that represents a space-optimized trie (prefix tree) in which
+each node that is the only child is merged with its parent.
+
+Under the hood, the Tree structure is managed with
+[elm-rosetree](https://package.elm-lang.org/packages/zwilias/elm-rosetree/latest/)
+
+In this package we have two kinds of implementations, one is ordered and the
+other is un-ordered. It's possible that the un-ordered isn't called "Radix tree"
```

Comment:
I think another pass at rewriting "un-ordered" as "unordered" everywhere would be nice